test/cpp/api/optim.cpp

```diff
@@ -452,3 +452,55 @@ TEST(OptimTest, AddParameter_LBFGS) {
 
   // REQUIRE this doesn't throw
 }
+
+bool double_comparison(const double a, const double b,
```

Comment:
It seems this function is only being used by check_lr_change, can we move this inside the function?